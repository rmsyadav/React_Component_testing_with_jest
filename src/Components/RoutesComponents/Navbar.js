import {NavLink} from 'react-router-dom';

const Navbar = ()=>{

    return(
    <>
        <nav className="navbar navbar-expand-lg navbar-warning bg-warning">
            <div className="container-fluid">
                <a
                href="https://reactjs.org"
                target="_blank"
                rel="noopener noreferrer"
                className='text-primary fw-bold'
                style={{textDecoration:"none"}}
              >
                <i className="bi bi-windows"></i><strong>Learn React</strong>
              </a>
                <button className="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
                <span className="navbar-toggler-icon"></span>
                </button>
                <div className="collapse navbar-collapse" id="navbarSupportedContent">
                <ul className="navbar-nav ms-auto mb-2 mb-lg-0">
                    <li className="nav-item">
                      <NavLink className="nav-link active fw-bold" to={"/"}  data-testid="home"><i className="bi bi-house-door-fill"></i>Home</NavLink>
                    </li>
                    <li className="nav-item">
                      <NavLink className="nav-link fw-bold" to={"/list"}  data-testid="list"><i className="bi bi-list-stars"></i>List</NavLink>
                    </li>
                    <li className="nav-item">
                      <NavLink className="nav-link fw-bold" to={"/signin"} data-testid="signin"><i className="bi bi-box-arrow-in-right"></i>SignIn</NavLink>
                    </li>
                </ul>
                </div>
            </div>
        </nav>
        
    </>
 )
}

export default Navbar;