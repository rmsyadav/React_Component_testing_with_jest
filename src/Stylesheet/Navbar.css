.navbar-img{
    background-image: url('../Assets/27620.png');
}