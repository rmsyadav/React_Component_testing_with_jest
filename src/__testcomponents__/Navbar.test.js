import {render,screen,act} from "@testing-library/react";
import renderer from 'react-test-renderer';
import userEvent from '@testing-library/user-event'
import Navbar from "../Components/RoutesComponents/Navbar";
import { BrowserRouter } from "react-router-dom";
import MyRoutes from "../Components/RoutesComponents/Router";

describe("App component testing:- ",()=>{

    it('Navbar component snapshot testing ',()=>{
  
      const navbarRoot = renderer.create(<BrowserRouter><Navbar></Navbar></BrowserRouter>).toJSON();
      expect(navbarRoot).toMatchSnapshot();
    })
  
    test('renders learn react link', async() => {
      const user = userEvent.setup()
      render(<BrowserRouter><MyRoutes></MyRoutes></BrowserRouter>);

      const linkElement = screen.getByTestId('home');
      expect(linkElement).toBeInTheDocument();
      await act(async()=>{
        await user.click(screen.getByTestId('home'));
      })
      const list= screen.getByText(/Todo App/i);
      expect(list).toBeInTheDocument();
      
    });

    test('renders learn react link', async() => {
        const user = userEvent.setup()
        render(<BrowserRouter><MyRoutes></MyRoutes></BrowserRouter>);
  
        const linkElement = screen.getByTestId('list');
        expect(linkElement).toBeInTheDocument();
        await act(async()=>{
          await user.click(screen.getByTestId('list'));
         })
        const list= screen.getByText(/Numbered List Group/i);
        expect(list).toBeInTheDocument();
        
      });
      test('checking wether redirecting on signin component', async() => {
        const user = userEvent.setup()
        render(<BrowserRouter><MyRoutes></MyRoutes></BrowserRouter>);
        const linkElement = screen.getByTestId('signin');
        expect(linkElement).toBeInTheDocument();
        await act(async()=>{
          await user.click(linkElement);
        })
        const signin= screen.getByText(/Please login to your account/i);
        expect(signin).toBeInTheDocument();
        
      });
    
  
  })