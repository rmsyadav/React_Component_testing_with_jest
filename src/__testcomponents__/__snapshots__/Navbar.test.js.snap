// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component testing:-  Navbar component snapshot testing  1`] = `
<nav
  className="navbar navbar-expand-lg navbar-warning bg-warning"
>
  <div
    className="container-fluid"
  >
    <a
      className="text-sucess"
      href="https://reactjs.org"
      rel="noopener noreferrer"
      target="_blank"
    >
      Learn React
    </a>
    <button
      aria-controls="navbarSupportedContent"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-bs-target="#navbarSupportedContent"
      data-bs-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="navbarSupportedContent"
    >
      <ul
        className="navbar-nav ms-auto mb-2 mb-lg-0"
      >
        <li
          className="nav-item"
        >
          <a
            aria-current="page"
            className="nav-link active active"
            data-testid="home"
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            data-testid="list"
            href="/list"
            onClick={[Function]}
          >
            List
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            aria-current="page"
            className="nav-link active"
            data-testid="signin"
            href="/"
            onClick={[Function]}
          >
            SignIn
          </a>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;
